how2meet/ui/pages/events.py

```diff
@@ -0,0 +1,139 @@
+
+        # Add navigation buttons
+        with ui.row().classes("w-full justify-center"):
+            ui.button("Back", on_click=lambda: ui.open("/"))
+            ui.button("New Event", on_click=lambda: ui.open(f"/events/new/{uuid.uuid4()}"))
+
+
+@router.page("/{event_id}")
+async def event_home(event_id: str):
+    """Detail page for a specific event"""
+    # Get the event from the API
+    # NOTE: This is a blocking call, so we use an async client
+
+    with frame("Event Home"):
+        async with httpx.AsyncClient() as client:
+            event = await client.get(f"http://localhost:8000/api/events/{event_id}", timeout=10)
+        try:
+            event = event.json()
+        except:
+            event = {}
+
+        # Display the event details
+        with ui.column().classes("w-full justify-between"):
+            for key, value in event.items():
+                ui.label(f"{key}: {value}")
+        ui.button("Back", on_click=lambda: ui.open("/events"))
+
+@router.page("/new/{event_id}")
+async def new_event(event_id: str):
+    """New event creation page
+
+    Args:
+        event_id (str): The unique ID of the event to be created.
+
+    Returns:
+        None
+    """
+
+
+    async def add_guest():
+        invite_uuid = str(uuid.uuid4())
+        # TODO add method to remove guest
+        with ui.card().classes("flex-grow position:relative") as card:
+            async def delete(invite_uuid=invite_uuid):
+                card.delete()
+            with ui.row().classes("flex-grow justify-between items-center"):
+                name_input = ui.input("Name")
+                email_input = ui.input("Email")
+                phone_input = ui.input("Phone Number")
+                ui.button("", icon="delete", color="red", on_click=lambda invite_uuid=invite_uuid: delete(invite_uuid)).classes("w-6 h-6")
+
+
+    with frame("New Event"):
+        with ui.row().classes("flex-grow justify-between"):
+            event_name_input = ui.input("Event Name")
+            event_location_input = ui.input("Location")
+        with ui.row():
+            user_input = ui.input("User")
+            password_input = ui.input("Password", password=True, password_toggle_button=True)
+        with ui.row():
+            with ui.expansion("Start Time"):
+                start_date_input = ui.date()
+                start_time_input = ui.time()
+            with ui.expansion("End Time"):
+                end_date_input = ui.date()
+                end_time_input = ui.time()
+            all_day_checkbox = ui.checkbox("All Day")
+        with ui.row().classes("w-full"):
+            description_input = ui.textarea("Description").classes("w-full")
+        with ui.row().classes("w-full"):
+            with ui.column().classes("w-full"):
+                add_guest_button = ui.button("Add guest", on_click=add_guest)
+
+        async def get_event_json():
```

Comment:
..this smells of something funky. Lots of duplicated effort. Instead, couldn't the `Event` model be imported and used directly here? For example:
```
start_date__input = ui.date()
start_time_input = ui.time()
```
-->
```
event = Event()
event.start_date = ui.date()
event.start_time = ui.time()
```

Reply:
No, because it's so pedantic that if everything isn't perfect then the validation will fail, so you can't do partial instantiations. Validation will be handled by the backend.

Reply:
Ugh that's annoying, I ran into something similar. What if you use the ORM model, not the Pydantic one? This just looks super frustrating to deal with as things change

Reply:
I've talked about this with both of you separately I believe, but to summarize: I wouldn't expect us to have to fit input data to a schema or type outside of individual field validators in the frontend. I would expect the client to simply send a JSON of data to the server, which is then responsible for validating the data and then calling a model create to create a DB entry.